src/condor_starter.V6.1/starter.cpp

```diff
@@ -2077,10 +2081,309 @@ Starter::jobEnvironmentCannotReady(int status, const struct UnreadyReason & urea
 		return 0;
 	}
 
+	// Ask the AP what to do.
+	std::ignore = daemonCore->Register_Timer(
+		0, (TimerHandlercpp) & Starter::requestGuidanceJobEnvironmentUnready,
+		"ask AP what to do",
+		this
+	);
+
+	return true;
+}
+
+/**
+ * After any file transfers are complete, will enter this method
+ * to setup anything else that needs to happen in the Starter
+ * before starting a job
+ *
+ * @return true
+ **/
+int
+Starter::jobEnvironmentReady( void )
+{
+	m_job_environment_is_ready = true;
+
+	// Ask the AP what to do.
+	std::ignore = daemonCore->Register_Timer(
+		0, (TimerHandlercpp) & Starter::requestGuidanceJobEnvironmentReady,
+		"ask AP what to do",
+		this
+	);
```

Comment:
`std::function` overload of `Register_Timer` ?

Reply:
Ayup.